src/main/java/step3/domain/WinningPrice.java

```diff
@@ -0,0 +1,52 @@
+package step3.domain;
+
+import java.util.Arrays;
+
+public enum WinningPrice {
+    FIRST_PRICE(6, 2_000_000_000),
+    SECOND_PRICE(5, 30_000_000),
+    THIRD_PRICE(5, 1_500_000),
+    FOURTH_PRICE(4, 50_000),
+    FIFTH_PRICE(3, 5_000),
+    OTHER(0, 0);
+
+
+    private static final int SECOND_PRIZE_CONDITION = 5;
+    private int winningNumbers;
+    private int price;
+
+
+    WinningPrice(int winningNumbers, int price) {
+        this.winningNumbers = winningNumbers;
+        this.price = price;
+    }
+
+    public boolean checkCount(int winningNumber) {
+        return winningNumbers == winningNumber;
+    }
+
+    public int getWinningNumbers() {
+        return winningNumbers;
+    }
+
+    public int getPrice() {
+        return price;
+    }
+
+    public static WinningPrice valueOf(int winningNumbers, boolean matchBonus) {
+
+        WinningPrice[] winningPrices = values();
+        if(winningNumbers == SECOND_PRIZE_CONDITION && matchBonus) {
+            return WinningPrice.SECOND_PRICE;
+        }
+
+        if(winningNumbers == SECOND_PRIZE_CONDITION && !matchBonus) {
+            return WinningPrice.THIRD_PRICE;
+        }
+
+        return Arrays.stream(winningPrices)
+                .filter(priceValue -> priceValue.checkCount(winningNumbers))
+                .findAny()
+                .orElse(OTHER);
+    }
```

Comment:
winningNumbers 라는 네이밍 보다는 당첨 갯수 같은 네이밍이 더 자연스러울 것 같아요 :)
```suggestion
    public static WinningPrice valueOf(int winningCount, boolean matchBonus) {
        if (winningCount== SECOND_PRIZE_CONDITION && matchBonus) {
            return WinningPrice.SECOND_PRICE;
        }

        if (winningCount== SECOND_PRIZE_CONDITION && !matchBonus) {
            return WinningPrice.THIRD_PRICE;
        }

        return Arrays.stream(WinningPrice.values())
                .filter(priceValue -> priceValue.checkCount(winningCount))
                .findAny()
                .orElse(OTHER);
    }
```